### ISSUE TITLE
CLAHE return nan on black image

### ISSUE THREAD

**Issue by YubinXie** [scikit-image/scikit-image#3773]
## Description
When I apply CLAHE to my images, some of which is np.zeros((n,n)) after thresholding, CLAHE gives errors nan, nan. I am thinking if it is more reasonable to return np.zeros(n.n)). Thanks!

## Way to reproduce
```python
import numpy as np
a=np.zeros((10,10))
img_rescale = exposure.equalize_adapthist(a, clip_limit=0.03)
plt.imshow(img_rescale,'gray')
print(img_rescale)
```

```python
array([[nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan]])

```

**Comment by stefanv** [comment]
Agreed.

**Comment by jni** [comment]
In #3402 I made lots of checks for `threshold_li` that could really extend to a bunch of methods, including CLAHE. We should probably pilfer all those checks and make a "sanity check this array" function.

**Comment by nandahkrishna** [comment]
Hello, can I contribute to this if it's still open?

**Comment by hmaarrfk** [comment]
Of course!!! It doesn't seem that anybody has referenced this issue in a PR.

Have you seen the contribution guide?
https://scikit-image.org/docs/stable/contribute.html

let us know if you need any more guidance at this point, somebody will surely chime in.

(i've seen this https://github.com/scikit-image/scikit-image/pull/2761 but it seems to have gotten stale)

**Comment by nandahkrishna** [comment]
Thanks!

**Comment by nandahkrishna** [comment]
The PR #2761 seems to be an enhancement, so shall I incorporate the same idea in a PR and simultaneously solve this issue? Or would it be best to just solve this alone?

My understanding of the issue is that any `nan` values in the output image must actually be zeroes. Do correct me if I'm wrong.

**Comment by hmaarrfk** [comment]
i think it is best to solve the nan problem in isolation of 2761. that helps ensure a faster review cycle.

my feeling is that it might be a more general bug, where by any constant matrix might output nans, but I think your investigation will confirm that.

**Comment by nandahkrishna** [comment]
My observation so far is that for any image of size less than (16,16) the function returns `nan`s. The input shape (16,16) gives a kernel size of (2,2) while anything lesser would give (1,1) (upto input shape (8,8)) - by the default method to get kernel size which is to divide by 8.

**Comment by nandahkrishna** [comment]
@hmaarrfk Would it be possible for you to suggest a fix? I'm not sure if I should change the algorithm to work for these smaller image sizes or raise some exception saying the size is too less.

**Comment by jni** [comment]
Fixed by #4015

**Comment by m-albert** [comment]
FYI:
#2761 has been updated and feedback was incorporated
